lib/rack/twilio_webhook_authentication.rb

```diff
@@ -17,18 +17,20 @@ module Rack
   # doesn't validate then the middleware responds immediately with a 403 status.
 
   class TwilioWebhookAuthentication
-    def initialize(app, auth_token, *paths)
+    def initialize(app, auth_token, *paths, &block)
       @app = app
       @auth_token = auth_token
+      define_singleton_method(:getAuthToken, block) if block_given?
       @path_regex = Regexp.union(paths)
     end
 
     def call(env)
       return @app.call(env) unless env["PATH_INFO"].match(@path_regex)
-      validator = Twilio::Util::RequestValidator.new(@auth_token)
       request = Rack::Request.new(env)
       original_url = request.url
       params = request.post? ? request.POST : {}
+      @auth_token ||=  getAuthToken(params['AccountSid'])
```

Comment:
Mixed thoughts for me here. My idea was to set the `@auth_token` with the dynamic value once we got it, and use the `@auth_token` for all practical purposes later on.

Rest assured that `get_auth_token(..)` will be called only once and that too if and only if the `@auth_token` is set `nil` during configuration.

All said, I've incorporated the feedback in my later commit because I felt when the other pair of eyes looking through the code is not comfortable, it's a sign of smell and I'm better of making the code more readable. 

Thanks for your time and feedback!